Dummy app updates

maxwondercorn opened the pull request.

rwwagner90 commented:
@maxwondercorn is this still WIP?

maxwondercorn commented:
@rwwagner90 re: the conflicts above.  I've already removed all the dummy app computed's with native getters or `@tracked` properties. 

After this commit is merged, you can review and merge the PR.  The `@classic` decorator can be removed once the addon is converted to glimmer components.  

Also, the new ember `font-awesome` add-on is incompatible with how the icons are used in ELT.  I think w3 could remove the font-awesome dependencies and still provide the same functionality.  It would be a breaking change and I'll dig into it a bit

rwwagner90 commented on tests/dummy/app/templates/components/materialize-icon.hbs:
```diff
@@ -1,3 +1,5 @@
 {{!-- BEGIN-SNIPPET materialize-icon --}}
-<i class="lt-sort-icon material-icons">{{get @sortIcons @sortIconProperty}}</i>
+<span>
```
Why did we wrap this in a span?

maxwondercorn replied:
The component was using `tagname: 'span'`

rwwagner90 commented on tests/dummy/app/templates/components/expanded-row.hbs:
```diff
@@ -7,9 +7,30 @@
     <div class="user-actions">
-      <a class="fa fa-envelope" href="mailto:{{@row.email}}"></a>
-      <a class="fa fa-facebook" href="#"></a>
-      <a class="fa fa-twitter" href="#"></a>
```
Why did we remove font-awesome here? I had updated font-awesome, so we should be able to still use it.

maxwondercorn replied:
New addon is official font-awesome add-on, not a community add-on. It does not support using class names. You need to use their `FaIcon` component or  the CDN.

I believe you can create a "kit' which is mini CDN but that requires managing something through their servers.  Long term we should try to remove the dependency.  Everyone using ELT and cuurent addon have both both addons in their project.

rwwagner90 replied:
@maxwondercorn yes, I know it is an official font-awesome addon, I updated to it. The class names should still work. I converted a bunch of `asc` and `desc` to `up` and `down` and they worked.

maxwondercorn replied:
That's interesting.  I had no icons displayed.  Maybe I still had asc/desc instead of up/down in the names but none of the others were showing either. IIRC my main app had the same issue and I switched to the fa component.

I'll pull all the changes back in and will undo changes as needed

rwwagner90 commented on tests/dummy/app/templates/components/rows/selectable-table.hbs:
```diff
@@ -1,40 +1,62 @@
+<div class='table-actions'>
   {{#if this.hasSelection}}
     <div
-      class="table-action fa fa-check-square"
```
Same with all the `fa` instances in this file. They should still work and we shouldn't change them.

rwwagner90 commented on tests/dummy/app/templates/components/scrolling-table.hbs:
```diff
@@ -2,16 +2,16 @@
     @iconSortable="fa fa-sort"
-    @iconAscending="fa fa-sort-up"
-    @iconDescending="fa fa-sort-down"
+    @iconAscending="fa fa-sort-asc"
```
These are wrong, they should be `up` and `down`. I changed these already to `up` and `down`.

maxwondercorn replied:
fixed

rwwagner90 reviewed (changes requested):
@maxwondercorn looks like a lot of font-awesome changes. We should put all that back to how it was, as I updated font-awesome to work already.

It seems like maybe there were some rebasing issues, as I already fixed several of these things and some got changed back.

maxwondercorn commented:
Hopefully the final pass.  Rebased and reverted font-awesome changes.  I still do not see icons in app but this can be addressed later.  All app examples work.

rwwagner90 commented:
@maxwondercorn I think some of the changes I made somehow made it into this PR. Perhaps a bad rebase or an accidental pull after rebasing before force pushing? Or perhaps an issue from me squashing and merging. Would you mind doing one final rebase and dropping the commits that should not be included in this PR please?

maxwondercorn commented:
Will look at it tomorrow. Very well could be a bad rebase. Wouldn’t be the first time.

rwwagner90 commented:
It does seem like font awesome icons are not showing up anymore, as you said. I don't know what changed since I opened my PR converting them. I will work on using the `<FaIcon>` component after we get this PR merged.

maxwondercorn commented:
Glad I'm not too crazy re: the icons 😄.  Yes it was a bad rebase.  Looks like I pulled partial changes from other merged PR's. Too much work to fix.  I just recreated the changes on the latest master and will close this PR. 

I'll have a new PR shortly

maxwondercorn commented:
Closing in preference of PR #793
